common/src/main/kotlin/io/homeassistant/companion/android/common/data/websocket/impl/WebSocketCoreImpl.kt

```diff
@@ -664,3 +737,98 @@ internal class WebSocketCoreImpl(
             .plus("api/websocket")
     }
 }
+
+/**
+ * Represents an active WebSocket message that has been sent and is awaiting a response.
+ *
+ * Stored in the active messages map to correlate incoming responses with their original requests.
+ */
+@VisibleForTesting
+internal sealed interface ActiveMessage {
+    /** Completes when the server sends a response for this message */
+    val responseDeferred: CompletableDeferred<RawMessageSocketResponse>
+
+    /**
+     * A one-shot message.
+     */
+    class SimpleActiveMessage(override val responseDeferred: CompletableDeferred<RawMessageSocketResponse>) :
+        ActiveMessage
+
+    /**
+     * A subscription that receive multiple events over time.
+     *
+     * @param responseDeferred Completes with the initial subscription confirmation response
+     * @param eventFlow The shared flow that emits events to external subscribers
+     * @param onEvent The channel used to send events into the flow
+     * @param request The original request, retained for resubscription after reconnection
+     */
+    class SubscriptionActiveMessage(
+        override val responseDeferred: CompletableDeferred<RawMessageSocketResponse>,
+        val eventFlow: SharedFlow<Any>,
+        val onEvent: Channel<Any>,
+        val request: WebSocketRequest,
+    ) : ActiveMessage
+}
+
+/**
+ * Represents a command to be sent through the send command channel for serialized WebSocket sending.
+ *
+ * @param T The type of result returned when the send operation completes
+ */
+private sealed interface CommandSender<T> {
```

Comment:
Usages like `sendMessage(CommandSender.WithAnswer.MessageCommandSender(request))` look foreign to me and I think it's because of the interface name and nesting. It is not a sender itself but an interface for commands so why not simply `Command.WithAnswer.Simple`/`Command.WithAnswer.Subscription`?